tests/test_common.js

```diff
@@ -512,3 +512,27 @@ test('testing timers are deleted correctly', t => {
     t.end()
   })
 })
+
+test('correct node behavior', t => {
+  const scope = nock('http://example.test')
+    .get('/')
+    .reply()
+
+  const reqSpy = sinon.spy()
+  const origHttpReq = http.request
+
+  http.request = reqSpy
+
+  http.get('http://example.test', res => {
+    t.equal(res.statusCode, 200)
+
+    res.on('data', reqSpy)
+
+    res.on('end', () => {
+      t.equal(reqSpy.called, false)
+      scope.done()
+      http.request = origHttpReq
```

Comment:
For safety, could you move this cleanup code into a `t.on('end', () => {})` callback?

Reply:
Yes, done